src/dataunion/Contracts.ts

```diff
@@ -53,11 +61,15 @@ export class Contracts {
     }
 
     getDataUnionSidechainAddress(mainnetAddress: EthereumAddress) {
-        if (!this.factorySidechainAddress) {
-            throw new Error('StreamrClient has no factorySidechainAddress configuration.')
+        if (!isAddress(this.factorySidechainAddress)) {
+            throw new Error('StreamrClient factorySidechainAddress configuration is ' + this.factorySidechainAddress ? 'missing' : 'not a valid Ethereum address')
         }
-        // NOTE! this must be updated when DU sidechain smartcontract changes: keccak256(CloneLib.cloneBytecode(data_union_sidechain_template))
-        const codeHash = '0x040cf686e25c97f74a23a4bf01c29dd77e260c4b694f5611017ce9713f58de83'
+
+        if (!isAddress(this.templateSidechainAddress)) {
+            throw new Error('StreamrClient dataUnionTemplateSidechainAddress configuration is ' + this.templateSidechainAddress ? 'missing' : 'not a valid Ethereum address')
```

Comment:
I think this ternary is around the wrong way, 

```suggestion
            throw new Error('StreamrClient dataUnionTemplateSidechainAddress configuration is ' + this.templateSidechainAddress ? 'not a valid Ethereum address' : 'missing')
```